driver/src/main/java/org/neo4j/driver/internal/pool/PooledConnection.java

```diff
@@ -156,12 +156,15 @@ public void close()
         {
             reset( StreamCollector.NO_OP );
             sync();
-            release.accept( this );
         }
         catch (Exception ex)
         {
             dispose();
         }
+        finally
+        {
+            release.accept( this );
```

Comment:
From the code, it looks like `release.accept` can fail with at least an `IllegalStateException`. Now this doesn't have a `catch` protecting it, what happens if/when this is thrown?